Squeeze #1655

suhy-jang opened the pull request:
- [ ] #1655

MarShaikh commented on ivy/functional/ivy/manipulation.py:
```diff
@@ -549,7 +549,26 @@ def squeeze(
+    >>> x = ivy.array([[[0], [1], [2]]])
+    >>> print(x.shape)
+    (1, 3, 1)
+    >>> print(ivy.squeeze(x))
```
Sorry for being nitpicky, but can you please add a single newline after every print output in the docstrings? This is because the way `ivy_test/test_docstrings.py` works is that it assumes a print statement output ends at a newline.

MarShaikh reviewed (changes requested):
The examples look great, but could you add them according to the [documentation](https://lets-unify.ai/ivy/deep_dive/11_docstring_examples.html)? Thanks!

suhy-jang commented:
I updated it to fit the docstring test format. 
Please review again @MarShaikh. Thank you.

MarShaikh reviewed (changes requested):
Hi, thanks for the updates. Could you include the examples within the functional, static, instance and other methods as well? They're missing from the docstrings. Like for example, the squeeze function has a definition in the `ivy/container/manipulation.py` as well as in the array. 
Can you also make sure that headings of each example is also exactly how they are in the [docs](https://lets-unify.ai/ivy/deep_dive/11_docstring_examples.html)? 
Thanks!

suhy-jang commented:
Thank you for your review.
I updated the docstring of the Squeeze function and modified the backend function.
@MarShaikh, please let me know if I need more work.

MarShaikh commented on ivy/functional/ivy/manipulation.py:
```diff
@@ -541,15 +541,103 @@ def squeeze(
+        a: ivy.array([0., 1., 2.]),
+        b: ivy.array([3., 4., 5.])
+    }
+
```
Hi, these examples look great, just that can you copy and paste them to ivy.Array.squeeze and ivy.Container.squeeze according to [array instance method and container instance method example](https://lets-unify.ai/ivy/deep_dive/11_docstring_examples.html#docstring-examples). Should be ready to merge then. Thanks.

suhy-jang replied:
Hi @MarShaikh. As you said, I have updated the instance methods of array and container. Since I do not have permission to merge, so could you please review and merge? Thank you in advance.

suhy-jang commented:
I copied the docstring example to ivy.Array.squeeze and ivy.Container.squeeze.
Please review again @MarShaikh. Thank you.

MarShaikh commented on ivy/functional/ivy/manipulation.py:
```diff
@@ -541,15 +541,103 @@ def squeeze(
+        b: ivy.array([3., 4., 5.])
+    }
+
+    Instance Method Examples
```
Hi, sorry to be so nitpicky, could you just remove the instance method examples from here, since you've already added them to container and arrays, these are redundant. Thanks.

suhy-jang commented:
I deleted the redundant examples from the main function, and please review them again @MarShaikh. 
Thank you so much for your attention and help.

MarShaikh reviewed (changes requested):
Could you please make sure the `test_docstrings.py` passes in the CI? I also noticed that you added the `out` argument in the backends. The out argument is only added if supported by the backend, which in the case of JAX, it isn't. Please see this for other backends as well. Thanks!

MarShaikh commented:
> Thank you for taking the time to review @MarShaikh. '`out` argument' was initially on my code, and it would have been more meaningful if included in the initial review. I updated it considering what you said, and now there is no problem with `test_docstrings.py` in the CI test related to `squeeze`. I am not familiar with this system yet, so if the problem of the failed test is still a problem in the following review, I would appreciate it if you could point out the exact cause and where it is. Thank you in advance, and please review it again.

Apologies for not pointing out the `out` argument earlier, in the .diff it seemed like something you added. Sorry the confusion again! Everything else looks great! Thanks for the contribution!
